tests/foreman/api/test_product.py

```diff
@@ -117,13 +110,21 @@ def test_positive_update_2(self, new_name):
         @Assert: Product Renamed to original
 
         """
-        old_name = self.product.name
-        # Update new product name
-        self.product.name = new_name
-        self.assertEqual(self.product.name, self.product.update(['name']).name)
-        # Rename product to old name and verify
-        self.product.name = old_name
-        self.assertEqual(self.product.name, self.product.update(['name']).name)
+        for new_name in generate_strings_list():
+            with self.subTest(new_name):
+                old_name = self.product.name
+                # Update new product name
+                self.product.name = new_name
+                self.assertEqual(
+                    self.product.name,
+                    self.product.update(['name']).name
+                )
+                # Rename product to old name and verify
+                self.product.name = old_name
+                self.assertEqual(
+                    self.product.name,
```

Comment:
I wonder if `old_name` should be used instead... Let's just say that something goes wrong with the update but no exceptions are raised? Then `self.product.name` would never be updated... but maybe I'm being paranoid hehehe